### ISSUE TITLE
ng serve does not support the "define" argument

### ISSUE THREAD

**Issue by Klaster1** [angular/angular-cli#28905]
### Command

serve

### Description

I want to pass environment variables into the application during serve and build. In Angular 19, this works for build: `ng build --define="GGCC_TARGET='hosted'"`, while this doesn't for serve: `ng serve --define="GGCC_TARGET='hosted'"`, with the following error:
```
Error: Unknown argument: define
```

### Describe the solution you'd like

The `--define` argument works for both serve and build commands.

### Describe alternatives you've considered

The custom "@ngx-env/builder:application" builder works, but I'd prefer to keep the amount of extra libraries to a minimum. Given how that builder is merely a thin wrapper around "@angular/build:application", I don't see why this couldn't work as I expect out of the box.

**Comment by alan-agius4** [comment]
This behavior is expected since the `define` configuration is derived from the `build` section of the `angular.json` file when executing `ng serve`.

```json
"architect": {
  "build": {
    "builder": "@angular-devkit/build-angular:application",
    "options": {
      "define": {
        "GGCC_TARGET": "'hosted'"
      },
```

**Comment by Klaster1** [comment]
Sorry, but I don't get it. Do you suggest to edit the "angular.json" instead of passing an env var, which worked for my team for years?

**Comment by alan-agius4** [comment]
Does the value of `GGCC_TARGET` need to change between various `ng serve` invocations where its value is set via an environment variable?

**Comment by Klaster1** [comment]
Yes, I pass different values to run the app in different configurations, i.e with some fractures hidden and different branding.
